src/Libraries/Microsoft.Extensions.Caching.Hybrid/Internal/TagSet.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.Caching.Hybrid.Internal;
+
+/// <summary>
+/// Represents zero (null), one (string) or more (string[]) tags, avoiding the additional array overhead when necessary.
+/// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1066:Implement IEquatable when overriding Object.Equals", Justification = "Equals throws by intent")]
+internal readonly struct TagSet
```

Comment:
Since there is a nice ToString implementation

```suggestion
[System.Diagnostics.DebuggerDisplay("{ToString()}")]
internal readonly struct TagSet
```